services/navigation/builtin/builtin.go

```diff
@@ -384,10 +408,16 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 				extra,
 			)
 			if err != nil {
+				svc.logger.Infof("hit motion error %s when navigating to waypoint %+v", err.Error(), wp)
 				return err
 			}
 
-			return svc.waypointReached(ctx)
+			if !success {
+				err := errors.New("failed to reach goal")
+				svc.logger.Infof("hit motion failed to reach goal %s when navigating to waypoint %+v", err.Error(), wp)
+				return err
```

Comment:
This function now reports failure to reach the goal as an error from the perspective of navigation.

Reply:
preference to wrap the error so that it reads "hit motion error when navigation to waypoint: {err}"

Reply:
done